gibbon-compiler/src/Gibbon/L3/Syntax.hs

```diff
@@ -81,7 +83,7 @@ data E3Ext loc dec =
                                    --   we'll probably represent (sizeof x) as (end_x - start_x) / INT
   | SizeOfScalar Var               -- ^ sizeof(var)
   | BoundsCheck Int Var Var        -- ^ Bytes required, region, write cursor
-  | BoundsCheckVector [(Int, Var, Var)] -- ^ Bytes required, region, write cursor but for a vector of cursors and regions
+  | BoundsCheckVector [(Int, Var, Var, (Var, Var))] -- ^ Bytes required, region, write cursor but for a vector of cursors and regions
```

Comment:
the comment doesn't seem to be updated for the new type. Also, a list of nested tuples is a deep nesting. Could you define some type synonyms? at least for the (Var, Var) part?